### PR TITLE
Whitelist of paths to push

### LINKED ISSUE #10: whitelisted cachix push and pins
Caches with limited storage could exhaust the storage if we push and pin every flake outputs of a subflake. Hence there should be an option to  do a whitelisted push and pin.

For example I only want to push and pin `omnix-cli` and `omnix-devshell` 

Screenshot 2024-11-07 at 4 07 18 PM

### PR DESCRIPTION
**Pull request by sinrohit**
Resolves #10 

Example command:

`nix run github:rsrohitsingh682/cachix-push/add-option-derivations -- --cache mycache --subflake omnix --prefix "$(git branch --show-current)" --names "omnix-cli omnix-devshell" < om.json`

This will push and pin only `omnix-cli` and `omnix-devshell` flake outputs.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -19,13 +19,18 @@ pog.pog {
       description = "The prefix to use in conjunction with system string when pinning paths";
       argument = "PREFIX";
     }
+    {
+      name = "derivations";
+      description = "The list of derivations to do whitelisted cachix push and pin";
+      argument = "DERIVATIONS";
+    }
```
Comment (srid) [inline default.nix]: Can we make this ... rigorous? I.e., if any of the names in the list are missing in omnix JSON (due to say developer typo), then the whole script should faily abruptly.

File: default.nix [hunk 2]
```diff
@@ -19,13 +19,18 @@ pog.pog {
       description = "The prefix to use in conjunction with system string when pinning paths";
       argument = "PREFIX";
     }
+    {
+      name = "derivations";
```
Comment (srid) [inline default.nix]:
```suggestion
      name = "names";
```

Since these refer to names of the derivations.

File: default.nix (hunk 1 above)
Comment (sinrohit) [reply default.nix]: I had already tested that, the script doesn't fail abruptly, it just skips that missing name.

Comment (srid) [reply default.nix]: By making it 'rigorous' I mean that the script should fail abruptly in those scenarios. Skipping (silently ignoring errors) means it is not rigorous.

Comment (sinrohit) [reply default.nix]: Got it. I thought just the opposite.

**Review by srid — APPROVED** [review]
LGTM, good job on the jq work.

If the jq script is going to become more complex in the future we should probably consider switching to something else like nushell.

**Pull request merged**